src/population_loader.rs

```diff
@@ -1,112 +1,109 @@
 use crate::parameters_loader::Parameters;
 use ixa::{
     context::Context,
-    define_derived_property, define_person_property, define_person_property_with_default,
+    define_person_property, define_person_property_with_default,
     error::IxaError,
     global_properties::ContextGlobalPropertiesExt,
     people::{ContextPeopleExt, PersonId},
 };
 
 use serde::Deserialize;
+use std::path::PathBuf;
 
 #[derive(Deserialize, Debug)]
 #[allow(non_snake_case)]
-pub struct PeopleRecord {
+pub struct PeopleRecord<'a> {
     age: u8,
-    homeId: usize,
+    homeId: &'a [u8],
 }
 
 define_person_property!(Age, u8);
 define_person_property!(HomeId, usize);
 define_person_property_with_default!(Alive, bool, true);
 
-#[allow(clippy::cast_possible_truncation)]
-#[allow(clippy::cast_sign_loss)]
-const CENSUS_MAX: usize = 1e15 as usize;
-
-#[allow(clippy::cast_possible_truncation)]
-#[allow(clippy::cast_sign_loss)]
-const CENSUS_MIN: usize = 1e14 as usize;
-
-define_derived_property!(CensusTract, usize, [HomeId], |home_id| {
-    if (CENSUS_MIN..CENSUS_MAX).contains(&home_id) {
-        home_id / 10_000
-    } else {
-        0 //Err(IxaError::IxaError(String::from("Census tract invalid from homeId")))
-    }
-});
+define_person_property!(CensusTract, usize);
 
 fn create_person_from_record(
     context: &mut Context,
     person_record: &PeopleRecord,
 ) -> Result<PersonId, IxaError> {
-    let person_id =
-        context.add_person(((Age, person_record.age), (HomeId, person_record.homeId)))?;
+    let tract: usize = String::from_utf8(person_record.homeId[..11].to_owned())
+        .expect("Home id should have 11 digits for tract + home id")
+        .parse()
+        .expect("Could not parse census tract");
+    let home_id: usize = String::from_utf8(person_record.homeId.to_owned())
+        .expect("Could not read home id")
+        .parse()
+        .expect("Could not read home id");
+
+    let person_id = context.add_person((
+        (Age, person_record.age),
+        (HomeId, home_id),
+        (CensusTract, tract),
+    ))?;
     Ok(person_id)
 }
 
+fn load_synth_population(context: &mut Context, synth_input_file: PathBuf) -> Result<(), IxaError> {
+    let mut reader = csv::Reader::from_path(synth_input_file).expect("Failed to open file.");
+    let mut raw_record = csv::ByteRecord::new();
+    let headers = reader.byte_headers().unwrap().clone();
```

Comment:
I didn't error propagate because IxaError doesn't have a way to convert from csv to IxaError. I was wondering whether we should do that here or in Ixa. But didn't think was appropriate for this skeleton.